server/src/main/java/org/opensearch/search/aggregations/metrics/NumericMetricsAggregator.java

```diff
@@ -78,6 +85,13 @@ public BucketComparator bucketComparator(String key, SortOrder order) {
             }
             return (lhs, rhs) -> Comparators.compareDiscardNaN(metric(lhs), metric(rhs), order == SortOrder.ASC);
         }
+
+        public CompositeIndexFieldInfo getSupportedStarTree() {
+            if (context.query() instanceof StarTreeQuery) {
+                return ((StarTreeQuery) context.query()).getStarTree();
+            }
+            return null;
+        }
```

Comment:
This and `getStarTreeValues` could also be moved into `StarTreeHelper`.

We shouldn't use inheritance as a form of code reuse. We're better off delegating to another class.